File: YeProfspilka.Backend/YeProfspilka.Core/Entities/Discount.cs
using YeProfspilka.Core.Entities.Base;
using YeProfspilka.Core.Enumerations;

namespace YeProfspilka.Core.Entities;

public class Discount : BaseEntity
{
    public string Name { get; set; }

    public bool? WithQrCode { get; set; }

    public bool? WithBarCode { get; set; }

    public Image? BarCodeImage { get; set; }

    public string? Description { get; set; }

    public DiscountType DiscountType { get; set; }
}

File: src/web/src/models/discount.ts
export enum DiscountType {
    AvailableForAll = 'availableForAll',
    AvailableForMembers = 'availableForMembers',
    OneTimeForAll = 'oneTimeForAll',
    OneTimeForMembers = 'oneTimeForMembers',
    OneTimeForFirstCourse = 'oneTimeForFirstCourse',
}

export interface Discount {
    id: string;
    name: string;
    withBarCode?: boolean;
    withQrCode?: boolean;
    withPromoCode?: boolean;
    barCodeImage: string;
    description: string;
    promoCode?: string;
    discountType: DiscountType;
    createdAtUtc: string;
}

export interface DiscountCode {
    id: string;
    discount: Discount;
    isActive: boolean;
    code: string;
    activateTimeUtc: string;
    deactivateTimeUtc: string;
}

export interface VerifyDiscountResult {
    fullName: string;
    email: string;
    image: string;
    discount: Discount;
    isSuccess: boolean;
}

export const buildVerifyUri = (dicountCode: DiscountCode): string | null => {
    if (window === undefined) return null;

    return `${window.location.origin}/discount/verify?discount=${dicountCode.discount.id}&discountCode=${dicountCode.code}`;
};
